### PR TITLE
test(auth.it.js): Fix failing auth integration test

### LINKED ISSUE #1203: Fix failing [sdk-auth] integration tests
### Description

Auth integration tests for `anonymous flow` and `customer password flow` are failing, see here for [details](https://circleci.com/gh/commercetools/nodejs/10624).

### Expected Behavior

All auth integration tests should pass.

### PR DESCRIPTION
**Pull request by Babanila**
#### Summary

Fix failing auth integration tests

#### Description
This PR fix the failing integration tests ([details](https://circleci.com/gh/commercetools/nodejs/10624)).
The `anonymous_id` and `customer_id` are needed in order to generate a `refresh token` which is a mandatory requirement to obtain an `access token`.
This is as a result of the newly added feature to the SDKs for external OAuth. For more details, see [here](https://github.com/commercetools/docs.commercetools.com/pull/1255).

#### Todo

- Tests
  - [x] Unit
  - [x] Integration

Closes #1203

### CODE REVIEW

File: integration-tests/sdk/auth.it.js [hunk 1]
```diff
@@ -23,6 +23,43 @@ function getApiClient(token) {
...
+  changeScopeToArrayOfObject = changeScopeToArrayOfObject.substr(
+    0,
+    changeScopeToArrayOfObject.length - 1
+  )
+  changeScopeToArrayOfObject += '}]'
+
+  // Convert to JSON
+  const modifiedScope = JSON.parse(changeScopeToArrayOfObject)
+
+  const deleteScopeParams = modifiedScope.map(scopeParams => {
```
Comment (Babanila) [inline integration-tests/sdk/auth.it.js]: This deletes the unused parameters.

File: integration-tests/sdk/auth.it.js [hunk 2]
```diff
@@ -23,6 +23,43 @@ function getApiClient(token) {
   })
 }
 
+function modifiedTokenInfo(tokenInput, unUsedParameter) {
+  const receivedScope = tokenInput.scope
+  let changeScopeToArrayOfObject = '[{'
```
Comment (Babanila) [inline integration-tests/sdk/auth.it.js]: Convert the scope received to an array of Object.

File: integration-tests/sdk/auth.it.js [hunk 3]
```diff
@@ -23,6 +23,43 @@ function getApiClient(token) {
...
+  // Convert to JSON
+  const modifiedScope = JSON.parse(changeScopeToArrayOfObject)
+
+  const deleteScopeParams = modifiedScope.map(scopeParams => {
+    if (scopeParams.hasOwnProperty(unUsedParameter))
+      delete scopeParams[unUsedParameter]
+    return scopeParams
+  })
+  const scopeValue = String(JSON.stringify(deleteScopeParams[0]))
+  const redefineScopeValue = scopeValue.split(/ |:|"/)
```
Comment (Babanila) [inline integration-tests/sdk/auth.it.js]: Change the scope back to its initial format.

File: integration-tests/sdk/auth.it.js [hunk 4]
```diff
@@ -23,6 +23,43 @@ function getApiClient(token) {
...
 
+function modifiedTokenInfo(tokenInput, unUsedParameter) {
+  const receivedScope = tokenInput.scope
+  let changeScopeToArrayOfObject = '[{'
+  const newScope = receivedScope.split(' ')
+
+  for (let i = 0; i < newScope.length; i++) {
+    const splitScope = newScope[i].split(':')
+    changeScopeToArrayOfObject +=
+      '"' + splitScope[0] + '":"' + splitScope[1] + '",'
```
Comment (adnasa) [inline integration-tests/sdk/auth.it.js]: unless we are using [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) + [Object.values](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object) + `JSON.stringify`, this is black magic 🔮

Comment (adnasa) [reply integration-tests/sdk/auth.it.js]: the point I'm making is that you should aim to incorporate a stategy of how you work with the scopes and have readable code. I encourage a combination of different native Array/Object functions to accomplish that

Comment (adnasa) [reply integration-tests/sdk/auth.it.js]: i'm confused.. this could be the same as `changeScopeToArrayOfObject = newScope[i]`

Comment (Babanila) [reply integration-tests/sdk/auth.it.js]: It's not black magic, check the code update.

File: integration-tests/sdk/auth.it.js [hunk 5]
```diff
@@ -104,7 +127,9 @@ describe('Auth Flows', () => {
 
       // check returned properties
       expect(tokenInfo).toHaveProperty('access_token')
-      expect(tokenInfo).toHaveProperty('scope', `manage_project:${projectKey}`)
+      expect(
+        modifiedTokenInfo(tokenInfo, ignoredResponseKeys[1])
```
Comment (adnasa) [inline integration-tests/sdk/auth.it.js]: so why are we running tests towards a function implemented _inside_ the test?
How does this fix integration test?

**Comment by daern91** [comment]
Hey @Babanila, let's look into this one together. If the tests are failing we should try to fix the bug in the source code. Not create a new feature inside the tests to make them green.

**Pull request merged**